duckduckgo/iOS
# Manage Subscription Settings

Task/Issue URL: https://app.asana.com/0/0/1206130131426002/f

**Description**:
- Integrate Subscription management options in settings
- Adds subscription Debug options to debug menu
- Minor fixes in SwiftUI Settings
- Base Restore Flow

**Test Requirements**:
- ⚠️ Test on device
- ⚠️ Requires App Store Sandbox - 
- ⚠️ Test `using DuckDuckGo-Alpha` Target

### Test Settings Options:

1. Purchase a Subscription via the sandbox
2. Once purchased, observe that the following subscription management (See image bewlo) options are available in Settings

![image](https://github.com/duckduckgo/iOS/assets/1156669/9f4819d2-1d86-48ca-b7f5-930b690a826f)

3. Tap "Manage Subscription"
4. Observe Subscription Details are shown


### Test AppStore Restore:
1. Go to Settings > All Debug Options > Subscriptions
2. Tap "Clear Authentication Data"
5. Tap "Sync App Store Account" enter the Sandbox Password and wait for confirmation
&. Observe Privacy Pro state is reset to Purchase Info as follows:

![image](https://github.com/duckduckgo/iOS/assets/1156669/fe43589c-8561-4346-952f-4224d76b3d9d)

6. Tap Learn More and try to subscribe again while you're still subscribed.  (Subscription duration is ~5 minutes)
7. You should see a "Subscription Found" message like this one:

![image](https://github.com/duckduckgo/iOS/assets/1156669/ca9a803a-5514-4727-91aa-626db4d7b385)

8. Tap "Restore"
9. You should be take to the subscription settings
10. Check Settings Menu, observe subscription management options are there again.
---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
